### PR TITLE
Changes from klarna master

### PR DESCRIPTION
**Pull request by PelleUllberg**

### CODE REVIEW

File: src/main/java/com/klarna/hiverunner/HiveServerContext.java [hunk 1]
```diff
@@ -19,13 +19,33 @@
...
+import java.util.Map;
+
 /**
  * Configuration for the HiveServer.
+ *
+ * Implementations of this interface should set the context of the HiveServer that is spawned by HiveRunner. {@link
+ * com.klarna.hiverunner.StandaloneHiveRunner} uses the {@link StandaloneHiveServerContext} to create a context with
+ * zero external dependencies.
+ *
+ * By implemeneting other contexts you may e.g. point hiveserver to a different metastore, pre installed external hadoop
```
Comment (Thoymas) [inline src/main/java/com/klarna/hiverunner/HiveServerContext.java]: Typo. Implementing.

File: src/main/java/com/klarna/hiverunner/StandaloneHiveServerContext.java [hunk 2]
```diff
@@ -16,36 +16,75 @@
...
 import java.io.File;
 import java.io.IOException;
+import java.util.Map;
 import java.util.UUID;
 
 import static org.apache.hadoop.hive.conf.HiveConf.ConfVars.*;
 
 /**
- * Abstract base class responsible for common configuration for running the HiveServer within this JVM with zero external dependencies.
+ * Abstract base class responsible for common configuration for running the HiveServer within this JVM with zero
```
Comment (Thoymas) [inline src/main/java/com/klarna/hiverunner/StandaloneHiveServerContext.java]: It's not abstract any more.

File: src/main/java/com/klarna/hiverunner/StandaloneHiveServerContext.java [hunk 3]
```diff
@@ -119,7 +167,22 @@ protected void configureSupportConcurrency(HiveConf conf) {
...
+
+        try {
+            Class.forName(JDBCDriver.class.getName());
+        } catch (ClassNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+
+        // Set the hsqldb driver
+        metaStorageUrl = "jdbc:hsqldb:mem:" + UUID.randomUUID().toString();
+        hiveConf.set("datanucleus.connectiondrivername", "org.hsqldb.jdbc.JDBCDriver");
```
Comment (Thoymas) [inline src/main/java/com/klarna/hiverunner/StandaloneHiveServerContext.java]: Why use the JDBCDriver.class.getName() in one place and a string constant in these two places. Stick to one way. I prefer the class.getName() myself.

File: src/main/java/com/klarna/hiverunner/config/HiveRunnerConfig.java [hunk 4]
```diff
@@ -134,28 +149,29 @@ public void setHiveExecutionEngine(String executionEngine) {
      */
     public void override(HiveRunnerConfig hiveRunnerConfig) {
         this.config.putAll(hiveRunnerConfig.config);
+        this.hiveConfSystemOverride.putAll(hiveRunnerConfig.hiveConfSystemOverride);
     }
 
     private String load(String property, String defaultValue,
-                        List<String> validValues) {
-        String value = load(property, defaultValue);
+                        List<String> validValues, Properties sysProperty) {
```
Comment (Thoymas) [inline src/main/java/com/klarna/hiverunner/config/HiveRunnerConfig.java]: typo. sysProperty => sysProperties

File: src/main/java/com/klarna/hiverunner/config/HiveRunnerConfig.java [hunk 5]
```diff
@@ -172,4 +188,19 @@ private int getInteger(String key) {
     private String getString(String key) {
         return (String) config.get(key);
     }
+
+    private Map<String, String> loadSystemConfig(Properties systemProperties) {
```
Comment (Thoymas) [inline src/main/java/com/klarna/hiverunner/config/HiveRunnerConfig.java]: method name. perhaps change to getHiveConfOverrideConfig ?

File: src/main/java/com/klarna/hiverunner/HiveServerContext.java (hunk 1 above)
Comment (PelleUllberg) [inline src/main/java/com/klarna/hiverunner/HiveServerContext.java]: Fixed

File: src/main/java/com/klarna/hiverunner/StandaloneHiveServerContext.java (hunk 2 above)
Comment (PelleUllberg) [inline src/main/java/com/klarna/hiverunner/StandaloneHiveServerContext.java]: Fixed

File: src/main/java/com/klarna/hiverunner/StandaloneHiveServerContext.java (hunk 3 above)
Comment (PelleUllberg) [inline src/main/java/com/klarna/hiverunner/StandaloneHiveServerContext.java]: Fixed
https://github.com/klarna/HiveRunner/commit/9925906aacbe386153377322eb40fa059cae1611

File: src/main/java/com/klarna/hiverunner/config/HiveRunnerConfig.java (hunk 4 above)
Comment (PelleUllberg) [inline src/main/java/com/klarna/hiverunner/config/HiveRunnerConfig.java]: Fixed
https://github.com/klarna/HiveRunner/commit/05ebebebc7de6a4853d13f3fd014c80b1c1541b4

File: src/main/java/com/klarna/hiverunner/config/HiveRunnerConfig.java (hunk 5 above)
Comment (PelleUllberg) [inline src/main/java/com/klarna/hiverunner/config/HiveRunnerConfig.java]: Fixed
https://github.com/klarna/HiveRunner/commit/f39f2f8a63329bb0415f496069f2cfd5e9d26a55

**Comment by Thoymas** [comment]
Approved.

**Pull request merged**